reconcile/dynatrace_token_provider/validate.py

```diff
@@ -0,0 +1,60 @@
+from collections.abc import Iterable
+
+from reconcile.gql_definitions.dynatrace_token_provider.token_specs import (
+    DynatraceTokenProviderTokenSpecV1,
+)
+
+
+class SecretNotUniqueError(Exception):
+    pass
+
+
+class TokenNameNotUniqueInSecretError(Exception):
+    pass
+
+
+class KeyNameNotUniqueInSecretError(Exception):
+    pass
+
+
+class TokenNameTooLongError(Exception):
+    pass
+
+
+MAX_TOKEN_NAME_LENGTH = 46
+
+
+def validate_token_specs(specs: Iterable[DynatraceTokenProviderTokenSpecV1]) -> None:
+    """
+    We cannot catch all potential errors through json schema definition.
+    """
+    for spec in specs:
+        seen_secrets: set[str] = set()
+        for secret in spec.secrets:
+            secret_definition_key = f"{secret.namespace}/{secret.name}"
+            if secret_definition_key in seen_secrets:
+                raise SecretNotUniqueError(
+                    f"A secret cannot be re-defined and must be unique per spec. Secret '{secret.name}' in namespace '{secret.namespace}' is defined multiple times in spec '{spec.name}'."
+                )
+            seen_secrets.add(secret_definition_key)
+
+            seen_tokens: set[str] = set()
+            seen_key_names: set[str] = set()
+            for token in secret.tokens:
+                if len(token.name) > MAX_TOKEN_NAME_LENGTH:
+                    raise TokenNameTooLongError(
+                        f"Token name '{token.name}' is too long. The maximum length is {MAX_TOKEN_NAME_LENGTH} characters."
+                    )
```

Comment:
size limit can be done by schema `maxLength`

Reply:
Oh nice! Created https://github.com/app-sre/qontract-schemas/pull/699 and removed token name length validation.